src/main/java/org/opensearch/sdk/BaseExtension.java

```diff
@@ -35,12 +36,37 @@ public abstract class BaseExtension implements Extension {
     protected ThreadPool threadPool;
 
     /**
-     * Empty constructor to fulfill abstract class requirements
+     * The extension settings include a name, host address, and port.
+     */
+    private ExtensionSettings settings;
+
+    /**
+     * Empty constructor to fulfill abastract class requirement
      */
     protected BaseExtension() {
 
     }
 
+    /**
+     * Instantiate this extension, initializing the connection settings and REST actions.
+     * @throws IOException on failure to load settings.
+     */
+    protected BaseExtension(String path) throws IOException {
+        try {
+            this.settings = Extension.readSettingsFromYaml(path);
+            if (settings == null || settings.getHostAddress() == null || settings.getHostPort() == null) {
+                throw new IOException("Failed to initialize Extension settings. No port bound.");
+            }
+        } catch (IOException e) {
+            throw new ExceptionInInitializerError(e);
+        }
+    }
+
```

Comment:
Please also add a constructor that takes an `ExtensionSettings` object and sets it directly.

Optionally you can change the `this.settings = ` in the other constructor to call the new one you created like `this(Extension.readSettingsFromYaml(path));`.

Reply:
@dbwiddis I add new constructor in `BaseExtension` which takes ExtensionSetting object and sets it directly. 
`protected BaseExtension(ExtensionSettings settings) {
        this.settings = settings;
    }`
Since the test class `ExtensionsRunnerForTest()` was using this empty constructor from `BaseExtension` in its constructor,  the `ExtensionsRunnerForTest()` Constructor need to add parameter. No sure we really need to pass the variable to the Constructor in `ExtensionsRunnerForTest()`.